server/internal/usecase/interfaces/request.go

```diff
@@ -42,6 +42,7 @@ type RequestFilter struct {
 type Request interface {
 	FindByID(context.Context, id.RequestID, *usecase.Operator) (*request.Request, error)
 	FindByIDs(context.Context, id.RequestIDList, *usecase.Operator) (request.List, error)
+	FindByItems(context.Context, id.ItemIDList, *usecase.Operator) (map[id.ItemID]request.List, error)
```

Comment:
```suggestion
	ItemStatus(context.Context, id.ItemIDList, *usecase.Operator) (map[id.ItemID]item.Status, error)
```

The part that calculates the status from the request is business logic. It should be handled at the use case or domain layer.